.github/workflows/windows-build-and-test.yaml

```diff
@@ -49,20 +57,89 @@ jobs:
-      # to linker failures when a target finished before another one
-      # it depends on.
-    - name: Build [${{ matrix.build_type }}]
-      run: cmake --build ${{ matrix.build_type }} --config ${{ matrix.build_type }}
-    - name: Install [${{ matrix.build_type }}]
-      run: cmake --install ${{ matrix.build_type }} --config ${{ matrix.build_type }}
-    - name: Create DB
-      run: ~/PostgreSQL/${{ matrix.pg }}/bin/initdb -U postgres -A trust
-    - name: Start PostgreSQL
-      run: ~/PostgreSQL/${{ matrix.pg }}/bin/pg_ctl start -o "-cshared_preload_libraries=timescaledb"
-    - name: Test CREATE EXTENSION
-      run: ~/PostgreSQL/${{ matrix.pg }}/bin/psql -U postgres -d postgres -c 'CREATE EXTENSION timescaledb'
+
+    - name: Configure TimescaleDB
+      run: cmake -B build_win -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} -DPG_PATH="$HOME/PostgreSQL/${{ matrix.pg }}" -DOPENSSL_ROOT_DIR="$HOME/PostgreSQL/${{ matrix.pg }}" -DASSERTIONS=ON
+    - name: Build TimescaleDB
+      run: cmake --build build_win --config ${{ matrix.build_type }}
+    - name: Install TimescaleDB
+      run: cmake --install build_win --config ${{ matrix.build_type }}
+
+    - name: Setup postgres cluster
+      if: matrix.pg != '12'
+      run: |
+        ~/PostgreSQL/${{ matrix.pg }}/bin/initdb -U postgres -A trust --locale=us --encoding=UTF8
+        mkdir -p ${{ env.TABLESPACE1 }}\_default
+        mkdir -p ${{ env.TABLESPACE2 }}\_default
+        icacls ${{ env.TABLESPACE1 }} /grant runneradmin:F /T
+        icacls ${{ env.TABLESPACE2 }} /grant runneradmin:F /T
+        copy build_win/test/postgresql.conf ${{ env.PGDATA }}
+        copy build_win/test/pg_hba.conf ${{ env.PGDATA }}
+        ~/PostgreSQL/${{ matrix.pg }}/bin/pg_ctl start -o "-cfsync=off"
+        ~/PostgreSQL/${{ matrix.pg }}/bin/psql -U postgres -d postgres -c 'CREATE USER root SUPERUSER LOGIN;'
+        ~/PostgreSQL/${{ matrix.pg }}/bin/psql -U postgres -d postgres -c 'SHOW log_filename;SHOW data_directory;SELECT version();'
+
+    - name: Setup WSL
+      if: matrix.pg != '12'
+      uses: Vampire/setup-wsl@v1
+      with:
+        additional-packages:
+          cmake
+          gcc
+          git
+          gnupg
+          make
+          postgresql-common
+    - name: Install postgres for test runner
+      if: matrix.pg != '12'
+      shell: wsl-bash {0}
+      run: |
+        yes | /usr/share/postgresql-common/pgdg/apt.postgresql.org.sh || true
+        apt-get install -y --force-yes postgresql-server-dev-${{ matrix.pg }}
+
+    - name: Run tests
+      if: matrix.pg != '12'
+      shell: wsl-bash {0}
+      run: |
+        export TEST_TABLESPACE1_PREFIX='${{ env.TABLESPACE1 }}'
+        export TEST_TABLESPACE2_PREFIX='${{ env.TABLESPACE2 }}'
+        cmake -B build_wsl -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} -DTEST_PGPORT_LOCAL=${{ env.PGPORT }}
+        make -C build_wsl regresschecklocal
+
+    - name: Setup postgres cluster for TSL tests
+      run: |
```

Comment:
Is it correct that this step should be performed with PG 12 (because the `Run TSL tests` step is excluded for PG 12)?